src/figuresClasses.ts

```diff
@@ -1,11 +1,107 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    const triangleSmallerSides = [a, b, c]
+      .filter((num) => num !== Math.max(a, b, c))
+      .reduce((cur, next) => cur + next, 0);
+
+    this.shape = 'triangle';
+    this.color = color;
+
+    if (Math.max(a, b, c) >= triangleSmallerSides) {
+      throw new Error(
+        'Longer side cannot be longer or equal of summury of smallest sides',
+      );
```

Comment:
The error message could be more descriptive about what went wrong. It would be better to say something like 'The longest side of a triangle must be less than the sum of the other two sides.'